File: tree/symmetric_tree.cpp
class Solution {
public:
    bool isSymmetric(TreeNode* root) {
        if (root == nullptr) return true;
        
        TreeNode *l = root;
        TreeNode *r = root;
        
        return isSym(l, r);

    }
    
    bool isSym(TreeNode *l, TreeNode *r) {
        // check if both are null
        if (!l || !r)
            return l == r; 
        if (l->val != r->val) 
            return false;
        else
            return isSym(l->left, r->right) && isSym(l->right, r->left);
    }
};

File: easy/tree/Symmetric_tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 * int val;
 * TreeNode left;
 * TreeNode right;
 * TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    // recursive
    public boolean isSymmetric(TreeNode root) {
        if (root == null) {
            return true;
        }
        return helper(root.left, root.right);

    }

    boolean helper(TreeNode left, TreeNode right) {
        if (left == null || right == null) {
            return (left == right);
        }
        if (left.val != right.val) {
            return false;
        }
        return (helper(left.left, right.right) && helper(left.right, right.left));
    }
}